org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/environments/ExecutionEnvironment.java

```diff
@@ -314,29 +279,20 @@ void initDefaultVM(IVMInstall vm) {
 		fDefault = vm;
 	}
 
-	/* (non-Javadoc)
-	 * @see org.eclipse.jdt.launching.environments.IExecutionEnvironment#getAccessRules(org.eclipse.jdt.launching.IVMInstall, org.eclipse.jdt.launching.LibraryLocation[], org.eclipse.jdt.core.IJavaProject)
-	 */
 	@Override
 	public IAccessRule[][] getAccessRules(IVMInstall vm, LibraryLocation[] libraries, IJavaProject project) {
 		IAccessRuleParticipant[] participants = getParticipants();
 		Map<IAccessRuleParticipant, IAccessRule[][]> rulesByParticipant = collectRulesByParticipant(participants, vm, libraries, project);
 		synchronized (this) {
 			Map<IAccessRuleParticipant, IAccessRule[][]> cachedRules = fParticipantMap.get(vm);
 			if (cachedRules == null || !cachedRules.equals(rulesByParticipant)) {
-				ArrayList<List<IAccessRule>> libLists = new ArrayList<>(); // array of lists of access rules
-				for (int i = 0; i < libraries.length; i++) {
-					libLists.add(new ArrayList<>());
-				}
-				for (int i = 0; i < participants.length; i++) {
-					IAccessRuleParticipant participant = participants[i];
+				List<List<IAccessRule>> libLists = IntStream.range(0, libraries.length) // array of lists of access rules
```

Comment:
I personally have to read the new code more than once to understand what it is supposed to do.
I'm not sure it is worth to refactor it to streams, as this is not a hotspot or whatever what would benefit from the new code, but reading is harder now (IMO).